axiom/tests/test_log_event.c

```diff
@@ -0,0 +1,359 @@
+  nr_log_event_set_guid(log, "test id 3");
+  nr_log_event_set_entity_name(log, "entity name here");
+  nr_log_event_set_hostname(log, "host name here");
+  tlib_pass_if_bool_equal("full log event", true,
+                          nr_log_event_to_json_buffer(log, buf));
+  nr_buffer_add(buf, NR_PSTR("\0"));
+  tlib_pass_if_str_equal(
+      "full log event",
+      "[{"
+      "\"message\":\"this is a test log error message\","
+      "\"log.level\":\"LOG_LEVEL_TEST_ERROR\","
+      "\"timestamp\":12345,"
+      "\"trace.id\":\"test id 1\","
+      "\"span.id\":\"test id 2\","
+      "\"entity.guid\":\"test id 3\","
+      "\"entity.name\":\"entity name here\","
+      "\"hostname\":\"host name here\""
+      "}]",
+      nr_buffer_cptr(buf));
+  nr_log_event_destroy(&log);
+
+  nr_buffer_destroy(&buf);
+}
+
+static void test_log_event_guid(void) {
+  nr_log_event_t* event = nr_log_event_create();
+
+  // Test : should not set a null guid
+  nr_log_event_set_guid(event, NULL);
+  tlib_pass_if_null("NULL guid", event->entity_guid);
+
+  // Test : should set the guid to an empty string
+  nr_log_event_set_guid(event, "");
+  tlib_pass_if_str_equal("empty string guid", "",
+                         event->entity_guid);
+
+  // Test : should set the guid
+  nr_log_event_set_guid(event, "wombat");
+  tlib_pass_if_str_equal("set the guid", "wombat",
+                         event->entity_guid);
+
+  // Test : One more set
+  nr_log_event_set_guid(event, "Kangaroo");
+  tlib_pass_if_str_equal("set a new guid", "Kangaroo",
+                         event->entity_guid);
+
+  nr_log_event_destroy(&event);
+}
+
+static void test_log_event_trace_id(void) {
+  nr_log_event_t* event = nr_log_event_create();
+
+  // Test : that is does not blow up when we give the setter a NULL pointer
+  nr_log_event_set_trace_id(event, NULL);
+  nr_log_event_set_trace_id(NULL, "wallaby");
+  tlib_pass_if_null("the trace should still be NULL",
+                    event->trace_id);
+
+  // Test : setting the trace id back and forth behaves as expected
+  nr_log_event_set_trace_id(event, "Florance");
+  tlib_pass_if_str_equal("should be the trace ID we set 1", "Florance",
+                         event->trace_id);
+  nr_log_event_set_trace_id(event, "Wallaby");
+  tlib_pass_if_str_equal("should be the trace ID we set 2", "Wallaby",
+                         event->trace_id);
+
+  nr_log_event_destroy(&event);
+}
+
+static void test_log_event_entity_name(void) {
+  nr_log_event_t* event = nr_log_event_create();
+
+  // Test : that is does not blow up when we give the setter a NULL pointer
+  nr_log_event_set_entity_name(event, NULL);
+  nr_log_event_set_entity_name(NULL, "wallaby");
+  tlib_pass_if_null("the entity_name should still be NULL",
+                    event->entity_name);
+
+  // Test : the getter should not blow up when we send it an event with a NULL
+  // entity name.
+  tlib_pass_if_null("NULL event -> NULL entity name", event->entity_name);
```

Comment:
Redundant

Reply:
thanks, addressed in https://github.com/newrelic/newrelic-php-agent/pull/490/commits/ea9a50a125729a1422335e5b182d676e19a953c5